src/tests/Microsoft.PowerFx.Core.Tests/TypeTests.cs

```diff
@@ -36,5 +37,34 @@ public void RecordTypeValidation()
 
             Assert.Equal(r1.GetHashCode(), r2.GetHashCode());
         }
+
+        [Theory]
+        [InlineData("Display1", true, "F1")]
+        [InlineData("F1", true, "F1")]
+        [InlineData("SomethingElse", false, null)]
+        public void TryGetFieldTypeLookupTest(string inputDisplayOrLogical, bool succeeds, string expectedLogical)
```

Comment:
>TryGetFieldTypeLookupTest [](http://example.com/codeflow?start=20&length=25)

One more case.... conflicts!
What a field's logical name conflicts with another fields display name. 

Logical=F1 , Display=Display1
Logical=Display1, Display=Display2

Reply:
Current Implementation prioritizes logical name.
Can't add the test case for the same coz, singleNameDisplayProvider handles the collision and would not let us add such records.